behavior/microservice/app.py

```diff
@@ -41,10 +42,122 @@ def infer(model_type, ou_type):
     return jsonify(Y)
 
 
+def connect():
+    # pylint: disable=assigning-non-slot
+    db = getattr(g, "__database", None)
+    if db is None:
+        db = g.__database = sqlite3.connect("inference.db")
+    return db
+
+
+@app.teardown_appcontext
+def close_connection(_):
+    db = getattr(g, "__database", None)
+    if db is not None:
+        db.close()
+
+
+def init_db():
+    with app.app_context():
+        db = connect()
+        db.cursor().execute(
+            """
+            CREATE TABLE IF NOT EXISTS inference_results (
+                query TEXT,
+                predicted_cost REAL,
+                true_cost REAL,
+                true_cost_valid INT,
+                action_state TEXT,
+                predicted_results TEXT)
+        """
+        )
+        db.commit()
+
+
+def get_inference_results(query):
+    # This by default returns the data sorted by query_id, then ordered
+    # by situations where true costing error'ed, and finally ordered by
+    # abs(true_cost - predicted_cost).
+    #
+    # Ordering by abs(true_cost - predicted_cost) gives us the instances
+    # where our prediction is way off first.
+    query_clause = "ORDER BY query, true_cost_valid, ABS(true_cost - predicted_Cost) DESC"
+    if query is not None:
+        query_clause = query
+
+    def dict_factory(cursor, row):
+        ret = {}
+        for idx, col in enumerate(cursor.description):
+            ret[col[0]] = row[idx]
+        return ret
+
+    db = connect()
+    db.row_factory = dict_factory
+    cursor = db.cursor().execute(
+        "SELECT *, ABS(true_cost - predicted_cost) as cost_diff FROM inference_results " + query_clause
+    )
+    result_set = cursor.fetchall()
+    db.commit()
+    return result_set
+
+
+@app.route("/")
+def index():
+    results = get_inference_results(None)
```

Comment:
Seems like you want a default argument on query. The default argument could be the default SQL that gets executed.